src/util/bson/bson.cpp

```diff
@@ -29,34 +29,39 @@ Bson::~Bson() noexcept
         case Type::String:
             string_val.~basic_string();
             break;
-        case Type::Document:
-            delete document_val;
-            document_val = nullptr;
-            break;
-        case Type::Array:
-            delete array_val;
-            array_val = nullptr;
-            break;
         case Type::Binary:
             binary_val.~vector<char>();
             break;
         case Type::RegularExpression:
             regex_val.~RegularExpression();
             break;
+        case Type::Document:
+            document_val.reset();
+            break;
+        case Type::Array:
+            array_val.reset();
+            break;
         default:
             break;
     }
 }
 
-Bson::Bson(const Bson& v) {
+Bson::Bson(const Bson& v) noexcept {
```

Comment:
This and the copy-assignment operator can throw exceptions as they allocate memory.